python/cudf_polars/cudf_polars/dsl/translate.py

```
@@ -27,7 +27,123 @@
+        :class:`expr.ErrorNode` nodes and collected in the the `errors` attribute.
+        After translation is complete, this list of errors should be inspected
+        to determine if the query is supported.
+        """
+        ctx: AbstractContextManager[None] = (
+            set_node(self.visitor, n) if n is not None else noop_context
+        )
+        # IR is versioned with major.minor, minor is bumped for backwards
+        # compatible changes (e.g. adding new nodes), major is bumped for
+        # incompatible changes (e.g. renaming nodes).
+        if (version := self.visitor.version()) >= (4, 0):
+            error = NotImplementedError(
+                f"No support for polars IR {version=}"
+            )  # pragma: no cover; no such version for now.
+            self.errors.append(error)  # pragma: no cover
+            raise error  # pragma: no cover
+
+        with ctx:
+            polars_schema = self.visitor.get_schema()
+            try:
+                node = self.visitor.view_current_node()
+            except Exception as e:
+                self.errors.append(e)
+                return ir.ErrorNode({}, str(e))
+            try:
+                schema = {k: dtypes.from_polars(v) for k, v in polars_schema.items()}
+            except Exception as e:
+                self.errors.append(e)
+                return ir.ErrorNode({}, str(e))
+            try:
+                result = _translate_ir(node, self, schema)
+            except Exception as e:
+                self.errors.append(e)
+                return ir.ErrorNode(schema, str(e))
+            if any(
+                isinstance(dtype, pl.Null)
+                for dtype in pl.datatypes.unpack_dtypes(*polars_schema.values())
+            ):
+                error = NotImplementedError(
+                    f"No GPU support for {result} with Null column dtype."
+                )
+                self.errors.append(error)
+                raise error
```

I raised the exception here instead of returning an `ErrorNode` because this test fails execute https://github.com/pola-rs/polars/blob/40f8f5d63c225cd3dcb4e220db1bd5622274b2ab/py-polars/tests/unit/operations/aggregation/test_horizontal.py#L415. You can see the error in this CI run https://github.com/rapidsai/cudf/actions/runs/11620082093/job/32361973152#step:8:5006